xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -785,8 +805,8 @@ bool CPVRChannelGroup::HasChangedChannels(void) const
   bool bReturn(false);
   CSingleLock lock(m_critSection);
 
-  for (PVR_CHANNEL_GROUP_MEMBERS::const_iterator it = m_members.begin(); !bReturn && it != m_members.end(); ++it)
-    bReturn = it->second.channel->IsChanged();
+  for (std::map<std::pair<int, int>, std::shared_ptr<PVRChannelGroupMember>>::const_iterator it = m_members.begin(); !bReturn && it != m_members.end(); ++it)
+    bReturn = it->second->channel->IsChanged();
```

Comment:
get rid of `bReturn`, do early `return true;` on match => range based for loop possible